README.md

```diff
@@ -123,19 +120,135 @@ hello world
 <!-- CODEBLOCK_END -->
 ```
 
-<!-- CODEBLOCK_END_EXAMPLE3 -->
+### Environment
+
+System environment is automatically passed to `command`s:
+
+<!--
+  CODEBLOCK_START
+  {
+    "type": "command",
+    "value": "echo \"My home directory is: $HOME\"",
+    "environment": {
+      "comment": "Be careful with these or you'll leak your actual home dir",
+      "HOME": "/Users/me"
+    }
+  }
+-->
+<!-- prettier-ignore -->
+~~~~~~~~~~bash
+$ echo "My home directory is: $HOME"
+
+My home directory is: /Users/me
+~~~~~~~~~~
+
+<!-- CODEBLOCK_END -->
+
+In some scenarios, passing system environment to `command`s may be undesirable. This functionality can be disabled using the [`--no-system-environment` CLI flag](#usage). This creates output such as:
+
+<!-- CODEBLOCK_START
+  {
+    "type": "command",
+    "value": "echo \"My password is: $MY_PASSWORD\"",
+    "environment": {
+      "comment": "just in case?",
+      "MY_PASSWORD": ""
+    }
+  }
+-->
+<!-- prettier-ignore -->
+~~~~~~~~~~bash
+$ echo "My password is: $MY_PASSWORD"
+
+My password is:
+~~~~~~~~~~
+
+<!-- CODEBLOCK_END -->
+
+Sometimes commands need a little extra nudging via environment to receive a usable output. Environment variables can be added using the `environment` key:
+
+<!-- CODEBLOCK_START_EXAMPLE_PASSED_ENVIRONMENT {"ignore": true} -->
```

Comment:
this threw me for a bit because we removed these comment fences earlier in the readme. those echos are ACTUALLY running, whereas this one isnt 👍